File: javaScript/invertBinaryTree.js
function invertBinaryTree(tree) {
  function invertNode(node) {
    let left = node.left;
    let right = node.right;

    node.left = right;
    node.right = left;

    if (node.left) {
      invertNode(node.left);
    }
    if (node.right) {
      invertNode(node.right);
    }
  }

  invertNode(tree);
  return tree;
}

// This is the class of the input binary tree.
class BinaryTree {
  constructor(value) {
    this.value = value;
    this.left = null;
    this.right = null;
  }
}

File: java/invertBinaryTree.java
public class InvertBinaryTree {
  public static void invertBinaryTree(BinaryTree tree) {
    BinaryTree left = tree.left;
    BinaryTree right = tree.right;

    tree.left = right;
    tree.right = left;

    if (tree.left != null) {
      invertBinaryTree(tree.left);
    }
    if (tree.right != null) {
      invertBinaryTree(tree.right);
    }
  }

  static class BinaryTree {
    public int value;
    public BinaryTree left;
    public BinaryTree right;

    public BinaryTree(int value) {
      this.value = value;
    }
  }
}
